server/public/plugin/hooks.go

```diff
@@ -284,4 +285,13 @@ type Hooks interface {
 	// config object can be returned to be stored in place of the provided one.
 	// Minimum server version: 8.0
 	ConfigurationWillBeSaved(newCfg *model.Config) (*model.Config, error)
+
+	// NotificationWillBePushed is invoked before a push notification is sent to the push
+	// notification server. The intention is to allow plugins to cancel a push notification.
+	//
+	// To reject a push notification, return an non-empty string describing why the notification
+	// was rejected.
+	//
+	// Minimum server version: 9.0
```

Comment:
To confirm, is this urgent for v9.0, or can it wait for v9.1?

Reply:
Ideally 9.0 so we can get proper mobile support sooner but it depends on how this review goes :)

Reply:
Yep, as mentioned, would be nice to have for v9.0. :)